contracts/meta-staking/src/contract.rs

```diff
@@ -1,55 +1,417 @@
+            execute::withdraw_delegator_reward(deps, env, info, validator)
+        }
+    }
+}
+
+mod execute {
+    use cosmwasm_std::{ensure, Coin, DistributionMsg, StakingMsg, SubMsg};
+
+    use crate::state::{ConsumerInfo, ValidatorInfo, CONSUMERS, VALIDATORS_BY_CONSUMER};
+
+    use super::*;
+
+    pub fn delegate(
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        validator: String,
+        amount: Coin,
+    ) -> Result<Response, ContractError> {
+        //// TODO converion rate? What to do if it changes?
+        // Check denom
+        let config = CONFIG.load(deps.storage)?;
+        if amount.denom != config.local_denom {
+            return Err(ContractError::IncorrectDenom {});
+        }
+
+        // Check this is a consumer calling this, fails if no consumer loads
+        let ConsumerInfo {
+            address: consumer_addr,
+            available_funds,
+            total_staked,
+        } = CONSUMERS.load(deps.storage, &info.sender)?;
+
+        // Validate validator address
+        let validator_addr = deps.api.addr_validate(&validator)?;
+
+        // HACK: We have the budget for each consumer funded by the x/gov module.
+        // A much better solution would be a generic liquid staking module.
+        // This is intended only for proof of concept
+        //
+        // Check consumer chain has available budget to delegate
+        ensure!(
+            available_funds + amount.amount > total_staked,
+            ContractError::NoFundsToDelegate {}
+        );
+
+        // HACK temporary work around for proof of concept. Real implementation
+        // would use something like a generic Superfluid module to mint or burn
+        // synthetic tokens.
+
+        // Update info for the (consumer, validator) map
+        // We add the amount delegated to the validator.
+        VALIDATORS_BY_CONSUMER.update(
+            deps.storage,
+            (&consumer_addr.clone(), &validator_addr.clone()),
+            |validator_info| -> Result<ValidatorInfo, ContractError> {
+                match validator_info {
+                    Some(validator_info) => Ok(ValidatorInfo {
+                        address: validator_info.address,
+                        consumer: validator_info.consumer,
+                        total_delegated: validator_info.total_delegated + amount.amount,
+                    }),
+                    // No one has delegated to this validator, we save the info
+                    // Initial amount is this delegation
+                    None => Ok(ValidatorInfo {
+                        address: validator_addr,
+                        consumer: consumer_addr,
+                        total_delegated: amount.amount,
+                    }),
+                }
+            },
+        )?;
+
+        // Subtract amount of available funds for that consumer
+        CONSUMERS.update(deps.storage, &info.sender, |current| match current {
```

Comment:
Just do a save.
You loaded it earlier, you save it here.

You could put the load and save together and do them in update if you wish.